### DISCUSSION TITLE
I should be able to compile `_middleware.ts` with dynamic code evaluation when I'm not using serverless functions

### DISCUSSION THREAD

**Discussion by ryall** [vercel/next.js#36446] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP Wed Mar 2 00:30:59 UTC 2022
    Binaries:
      Node: 16.14.2
      npm: 8.5.0
      Yarn: 1.22.17
      pnpm: N/A
    Relevant packages:
      next: 12.1.6-canary.5
      react: 17.0.2
      react-dom: 17.0.2

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

While running `next build` I get the **breaking error**:

```
../shared/node_modules/lodash/lodash.js
Dynamic Code Evaluation (e. g. 'eval', 'new Function') not allowed in Middleware pages/_middleware
```

I understand this has been discussed before, and I understand it's to provide compatibility with Vercel edge functions - but I don't really need any of that and I'm currently unable to build my project because of this error. 

I think it just needs a config option to be able to disable this restriction as it does not apply to everyone.

### Expected Behavior

I should be able to compile `_middleware.ts` with dynamic code evaluation when I'm not using serverless functions.

### To Reproduce

I'm simply using a few different `lodash` functions, like `find` etc, and this error appears.

**Downgrading to `12.0.10` fixes the issue.** This issue was introduced since `12.1.0`.

**Comment by magicspon** [comment]
i'm seeing this error as well.

```
Dynamic Code Evaluation (e. g. 'eval', 'new Function') not allowed in Middleware pages/_middleware
```

everything was working as expected, until I enabled cross package imports (i'm working in a mono repo)

```javascript
const { join } = require('path')

const workspace = join(__dirname, '..')

/**
 * @type {import('next').NextConfig}
 */
module.exports = {
	poweredByHeader: false,

	eslint: {
		dirs: ['src'],
	},

	webpack: (config, options) => {
		config.module = {
			...config.module,
			rules: [
				...config.module.rules,
				{
					test: /\.(js|jsx|ts|tsx)$/,
					include: [workspace],
					exclude: /node_modules/,
					use: options.defaultLoaders.babel,
				},
			],
		}

		return config
	},
}

```

If i remove the webpack config, the build works as expected.
